geps/gep-2257.md

```diff
@@ -0,0 +1,105 @@
+# GEP-2257: Gateway API Duration Format
+
+* Issue: [#2257](https://github.com/kubernetes-sigs/gateway-api/issues/2257)
+* Status: Experimental
+
+## TL;DR
+
+As we extend the Gateway API to have more functionality, we need a standard
+way to represent duration values. The first instance is the [HTTPRoute
+Timeouts GEP][GEP-1742]; doubtless others will arise.
+
+[GEP-1742]:/geps/gep-1742
+
+## Gateway API Duration Format
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 8174].
+
+[RFC 8174]:https://datatracker.ietf.org/doc/html/rfc8174
+
+A _Gateway API Duration_ or _GEP-2257 Duration_ is a string value that:
+
+- MUST match the regular expression `^([0-9]{1,5}(h|m|s|ms)){1,4}$` and
+- MUST be interpreted as specified by [Golang's `time.ParseDuration`][gotime].
+
+Since both of these conditions MUST be true, the effect is that GEP-2257
+Durations are a subset of what `time.ParseDuration` supports:
+
+- A GEP-2257 Duration MUST be one to four _components_, each of which consists
+  of an integer _value_ followed immediately by a _unit_. For example, `1h`,
+  `5m`, `1h5m`, and `1h30m30s500ms` are all valid GEP-2257 Durations.
+
+- For each component, the value MUST be one to five decimal digits. Floating
+  point is not allowed. Leading zeroes do not mean octal; the value MUST
+  always be interpreted as a decimal integer. For example, `1h`, `60m`, `01h`,
+  and `00060m` are all equivalent GEP-2257 Durations.
+
+- For each component, the unit MUST be one of `h` (hour), `m` (minute), `s`
+  (second), or `ms` (millisecond). No units larger than hours or smaller than
+  milliseconds are supported.
+
+- The total duration expressed by a GEP-2257 Duration string is the sum of
+  each of its components. For example, `1h30m` would be a 90-minute duration,
+  and `1s500ms` would be a 1.5-second duration.
+
+- There is no requirement that all units must be used. A GEP-2257 Duration of
+  `1h500ms` is supported (although probably not terribly useful).
+
+- Units MAY be repeated, although users SHOULD NOT rely on this support since
+  this GEP is Experimental, and future revisions may remove support for
+  repeated units. If units are repeated, the total duration remains the sum of
+  all components: a GEP-2257 duration of `1h2h20m10m` is a duration of 3 hours
+  30 minutes.
+
+- Since the value and the unit are both required within a component, `0` is
+  not a valid GEP-2257 duration string (though `0s` is). Likewise the empty
```

Comment:
Do we want to say anything about conventions and normalization?

e.g.
* While 0h, 0m, 0s, and 0ms are equivalent, the conventional representation for 0 is "0s".
* The conventional "normal form" representation for a duration populates units in descending order of size.  For example "90m" and "30m1h" both normalize to "1h30m".

Reply:
Yeah, I like this idea. Added.